Device Service panic when core-metadata host name is incorrect

If you assign a wrong metadata hostname, the following panic error will happen

Couldn't register to metadata service: Get http://edgex-core-metadata:48081/api/v1/deviceservice/name/device-simple: dial tcp: lookup edgex-core-metadata: no such host
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x7506bb]

goroutine 1 [running]:
github.com/edgexfoundry/device-sdk-go/internal/autoevent.(*manager).StopAutoEvents(0x0)
	C:/Development/EdgeX/device-sdk-go/internal/autoevent/manager.go:53 +0x5b
github.com/edgexfoundry/device-sdk-go/pkg/service.(*Service).Stop(0xc000186100, 0xc000026700)
	C:/Development/EdgeX/device-sdk-go/pkg/service/service.go:67 +0x5e
github.com/edgexfoundry/device-sdk-go/pkg/service.Main(0x8c525c, 0xd, 0x8cbfc7, 0x1a, 0x96e860, 0xc0000791a0, 0x96b7a0, 0xc0000267c0, 0xc000051240, 0xc0000ea000, ...)
	C:/Development/EdgeX/device-sdk-go/pkg/service/main.go:74 +0x674
github.com/edgexfoundry/device-sdk-go/pkg/startup.Bootstrap(0x8c525c, 0xd, 0x8cbfc7, 0x1a, 0x96e860, 0xc0000791a0)
	C:/Development/EdgeX/device-sdk-go/pkg/startup/bootstrap.go:20 +0x119
main.main()
	C:/Development/EdgeX/device-sdk-go/example/cmd/device-simple/main.go:23 +0x7b

--- a/internal/autoevent/manager.go
+++ b/internal/autoevent/manager.go
@@ -50,11 +50,15 @@ func (m *manager) StartAutoEvents() {
 
 func (m *manager) StopAutoEvents() {
 	mutex.Lock()
-	for k, v := range m.execsMap {
-		for _, e := range v {
-			e.Stop()
+	// because service.Stop might be called before autoevent manager is created
+	// in bootstrap, we need this check to prevent nil pointer dereference.
+	if m != nil {
+		for k, v := range m.execsMap {
+			for _, e := range v {
+				e.Stop()
+			}
+			delete(m.execsMap, k)
 		}
-		delete(m.execsMap, k)
 	}
 	mutex.Unlock()
 }

